src/msg/async/AsyncMessenger.cc

```diff
@@ -788,24 +788,21 @@ void AsyncMessenger::shutdown_connections(bool queue_reset)
 
   for (const auto& [e, c] : conns) {
     ldout(cct, 5) << __func__ << " mark down " << e << " " << c << dendl;
-    c->get_perf_counter()->dec(l_msgr_active_connections);
     c->stop(queue_reset);
   }
   conns.clear();
 
   for (const auto& c : anon_conns) {
     ldout(cct, 5) << __func__ << " mark down " << c << dendl;
-    c->get_perf_counter()->dec(l_msgr_active_connections);
     c->stop(queue_reset);
   }
   anon_conns.clear();
 
   {
     std::lock_guard l{deleted_lock};
-    if (cct->_conf->subsys.should_gather<ceph_subsys_ms, 5>()) {
```

Comment:
i suspect this was an optimisation . and it was introduced by https://github.com/ceph/ceph/pull/27324/files#diff-e578bc991aa81a83d1fb726bdb179dfb966188bd4e0ae5dc42ed5e40551a7c37R806 . @batrick any reason we should preserve this ?

Reply:
no particular reason